code/modules/mob/dead/observer/observer.dm

```diff
@@ -782,6 +782,35 @@ This is the proc mobs get to turn into a ghost. Forked from ghostize due to comp
 	if(isobserver(user) && check_rights(R_SPAWN))
 		change_mob_type( /mob/living/carbon/human , null, null, TRUE) //always delmob, ghosts shouldn't be left lingering
 
+/mob/dead/observer/proc/ondemand_possess(mob/living/target)
+	var/possessable = target.ondemand_possessable
+	if(!isnum(possessable) || possessable <= 0)
```

Comment:
same here, you only need `if(target.on_demand_possessable)`

Reply:
this isn't a boolea , it's a number. what if this is -1?